dartagnan/src/main/java/com/dat3m/dartagnan/wmm/utils/Arch.java

```diff
@@ -1,7 +1,7 @@
 package com.dat3m.dartagnan.wmm.utils;
 
 public enum Arch {
-    NONE, ARM, ARM8, POWER, TSO;
+    ARM, ARM8, NONE, POWER, TSO;
```

Comment:
Did you reorder the options alphabetically? I find NONE as a first option a lot clearer cause it has a special meaning and is the default value in most cases.

Reply:
Since the `Selector` ordered them using the enumeration order, I did this.
However I now make that the `Selector` class order whatever collection of elements we use.

Reply:
Hmm, the ```Selector``` ordered them according to enumeration order? If so, why don't we use NONE first? It sounds to me like you want an alphabetical order in the ```Selector``` but isn't the order as we have it in the enum better? NONE first sounds like the best option, especially if we default to it.

Reply:
When the selector is expanded, the options are shown in the order coming from the array that is used to create the selector. This order coincides with the enum order. I wanted to display the order alphabetically because I think it is nicer to the user, in particular when there are quite a few options.

Unless specified (which we do), the selector selects by default the first option.
I modified the selector created to do the ordering here (this implies that now also the solvers get ordered). I also implement all enum classes we use as options in a consistent way and added a `getDefault` method do now it is clear in each class which is the default option (before this info was done during the parsing or UI panels creation).

Reply:
I understand that ```Selector``` uses the order of the array. But I'm wondering why the option NONE should be alphabetically ordered with all the other architectures. Our initial enum layout was NONE and then the Architectures in alphabetical order. I would say this layout is ideal, cause it puts the special NONE value first (which also happens to be the default) and then orders the remaining one's alphabetically. I would use just that order for the user as well, cause having the common value NONE all over the place (it might be at very different places for different options) feels wrong to me. NONE should probably the first item for all ```Selectors```, even for the Alias analysis.

Reply:
I don't think relying in the order that enums are defined is good, this I added a method that returns the intended order. I also create a helper class which handles the order for `Solvers`. As you suggested, for options having a `NONE`, I put this first in the order and the rest based on alphabetical order of they `toString()` representation.